cluster/gce/cluster.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A library with any cluster level operations during kube-up, kube-down etc.
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${KUBE_ROOT}/cluster/gce/common.sh"
+
+# Provisions cluster level resources such as network and firewall rules.
+function cluster::provision() {
+  detect-project
+
+  if ! gcloud compute networks --project "${PROJECT}" describe "${NETWORK}" &>/dev/null; then
+    echo "Creating new network: ${NETWORK}"
+    # The network needs to be created synchronously or we have a race. The
+    # firewalls can be added concurrent with instance creation.
+    gcloud compute networks create --project "${PROJECT}" "${NETWORK}" --range "10.240.0.0/16"
+  fi
+
+  if ! gcloud compute firewall-rules --project "${PROJECT}" describe "${NETWORK}-default-internal" &>/dev/null; then
+    echo "Creating firewall rule for internal traffic"
+    gcloud compute firewall-rules create "${NETWORK}-default-internal" \
```

Comment:
There's a function `create-firewall-rule` that could be used here (maybe in a follow-up PR if you want this one to just be moving stuff around).